samples/python/kinetics/extract_submechanism.py

```diff
@@ -50,6 +51,9 @@
     if not all(product in species_names for product in R.products):
         continue
 
+    if R.third_body_collider not in colliders:
+        continue
+
```

Comment:
I would like it if we could avoid pushing the responsibility for this filtering step onto the user.